rts/lra/lra-proxy/api/src/main/java/io/narayana/lra/client/internal/proxy/nonjaxrs/LRAParticipantResource.java

```diff
@@ -0,0 +1,99 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+
+import static io.narayana.lra.LRAConstants.COMPENSATE;
+import static io.narayana.lra.LRAConstants.COMPLETE;
+import static io.narayana.lra.LRAConstants.FORGET;
+import static io.narayana.lra.LRAConstants.STATUS;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_PARENT_CONTEXT_HEADER;
+
+@ApplicationScoped
+@Path(LRAParticipantResource.RESOURCE_PATH)
+public class LRAParticipantResource {
+
+    public static final String RESOURCE_PATH = "lra-participant-proxy";
+
+    private LRAParticipantRegistry lraParticipantRegistry = LRAParticipantRegistry.getInstance();
+
+    @PUT
+    @Path("{participantId}/" + COMPENSATE)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Compensate
+    public Response compensate(@PathParam("participantId") String participantId,
+                               @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId,
+                               @HeaderParam(LRA_HTTP_PARENT_CONTEXT_HEADER) String parentId) {
+        return lraParticipantRegistry.getParticipant(participantId).compensate(createURI(lraId), createURI(parentId));
+    }
+
+    @PUT
+    @Path("{participantId}/" + COMPLETE)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Complete
+    public Response complete(@PathParam("participantId") String participantId,
+                             @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId,
+                             @HeaderParam(LRA_HTTP_PARENT_CONTEXT_HEADER) String parentId) {
+        return lraParticipantRegistry.getParticipant(participantId).complete(createURI(lraId), createURI(parentId));
+    }
+
+    @GET
+    @Path("{participantId}/" + STATUS)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Status
+    public Response status(@PathParam("participantId") String participantId,
+                           @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId,
+                           @HeaderParam(LRA_HTTP_PARENT_CONTEXT_HEADER) String parentId) {
+        return lraParticipantRegistry.getParticipant(participantId).status(createURI(lraId), createURI(parentId));
+    }
+
+    @DELETE
+    @Path("{participantId}/" + FORGET)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Forget
+    public Response forget(@PathParam("participantId") String participantId,
+                           @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId,
+                           @HeaderParam(LRA_HTTP_PARENT_CONTEXT_HEADER) String parentId) {
+        return lraParticipantRegistry.getParticipant(participantId).forget(createURI(lraId), createURI(parentId));
+    }
+
+    private URI createURI(String value) {
+        return value != null ? URI.create(value) : null;
```

Comment:
these values will be passed to the actual participant invocations so they should be null if they are not received by the invocation (they are taken from the header values).

Reply:
The participant will not be invoked because
```java
URI.create(value)
```
with throw a null pointer which will be caught by resteasy and converted to a generic 500 code

Reply:
My last comment was invalid, thanks for the answer. I will resolve this converstation.